go/vt/vttablet/tabletmanager/rpc_replication.go

```diff
@@ -463,23 +468,17 @@ func (tm *TabletManager) demotePrimary(ctx context.Context, revertPartialFailure
 	}
 
 	// Now that we know no writes are in-flight and no new writes can occur,
-	// set MySQL to read-only mode. If we are already read-only because of a
+	// set MySQL to super_read_only mode. If we are already super_read_only because of a
 	// previous demotion, or because we are not primary anyway, this should be
 	// idempotent.
-	if setSuperReadOnly {
-		// Setting super_read_only also sets read_only
-		if err := tm.MysqlDaemon.SetSuperReadOnly(true); err != nil {
-			if strings.Contains(err.Error(), mysql.ERUnknownSystemVariable.ToString()) {
-				log.Warningf("server does not know about super_read_only, continuing anyway...")
-			} else {
-				return nil, err
-			}
-		}
-	} else {
-		if err := tm.MysqlDaemon.SetReadOnly(true); err != nil {
+	if _, err := tm.MysqlDaemon.SetSuperReadOnly(true); err != nil {
```

Comment:
Should we not be doing this in a defer call to make sure it is always executed?

Reply:
I am afraid if I put it in defer call then even in case of partial failures (due to which we won't able to demote) we will still  turn it to super_read_only to 'ON'. 
Furthermore if you see the defer call down handles the same scenario where in case of partial failures or any erro (FinalErr) it sets ReadOnly to 'False'. This is to make sure we don't end up changing the state in case demotePrimary fails for any reason.

Reply:
This sounds good.